router/pkg/metric/metric_store.go

```diff
@@ -141,6 +159,19 @@ func NewStore(opts ...Option) (Store, error) {
 	return h, nil
 }
 
+// setCardinalityLimit sets the cardinality limit for open telemetry.
+// This feature is experimental in otel-go and may be exposed in a different way in the future.
+// In order to avoid creating a large number of metric streams, we set a hard limit that can be collected for a single instrument.
+func setCardinalityLimit(limit int) error {
+	if limit < 0 {
+		// setting a limit of 0 disables the cardinality limit
+		limit = 0
```

Comment:
Why should prevent someone from disabling it at all? I think we should just support the full API and only enforce a default limit.